accessor/cuda_helper.hpp

```diff
@@ -17,7 +17,15 @@
 #include "utils.hpp"
 
 
+struct __half;
+
+
```

Comment:
There are some ABI/name mangling differences between struct and class in MSVC - are you sure this will be defined as a struct always?

Also for cleaner headers/exports, maybe we should make this conditional on CUDA compilation?